test/src/org/apache/jmeter/protocol/http/util/TestHTTPArgument.java

```diff
@@ -19,79 +19,199 @@
-            assertEquals("name.?", clone.getName());
-            assertEquals("value_ here", clone.getValue());
-        }
-
-        @Test
-        public void testConversion() throws Exception {
-            Arguments args = new Arguments();
-            args.addArgument("name.?", "value_ here");
-            args.addArgument("name$of property", "value_.+");
-            HTTPArgument.convertArgumentsToHTTP(args);
-            CollectionProperty argList = args.getArguments();
-            HTTPArgument httpArg = (HTTPArgument) argList.get(0).getObjectValue();
-            assertEquals("name.%3F", httpArg.getEncodedName());
-            assertEquals("value_+here", httpArg.getEncodedValue());
-            httpArg = (HTTPArgument) argList.get(1).getObjectValue();
-            assertEquals("name%24of+property", httpArg.getEncodedName());
-            assertEquals("value_.%2B", httpArg.getEncodedValue());
-        }
-        
-        @Test
-        public void testEncoding() throws Exception {
-            HTTPArgument arg;
-            arg = new HTTPArgument("name.?", "value_ here", false);
-            assertEquals("name.?", arg.getName());
-            assertEquals("value_ here", arg.getValue());
-            assertEquals("name.%3F", arg.getEncodedName());
-            assertEquals("value_+here", arg.getEncodedValue());
-            // Show that can bypass encoding:
-            arg.setAlwaysEncoded(false);
-            assertEquals("name.?", arg.getEncodedName());
-            assertEquals("value_ here", arg.getEncodedValue());
-            
-            // The sample does not use a valid encoding
-            arg = new HTTPArgument("name.?", "value_ here", true);
-            assertEquals("name.?", arg.getName());
-            assertEquals("value_ here", arg.getValue());
-            assertEquals("name.%3F", arg.getEncodedName());
-            assertEquals("value_+here", arg.getEncodedValue());
-            arg.setAlwaysEncoded(false); // by default, name/value are encoded on fetch 
-            assertEquals("name.?", arg.getEncodedName());
-            assertEquals("value_ here", arg.getEncodedValue());
-            
-            // Try a real encoded argument
-            arg = new HTTPArgument("name.%3F", "value_+here", true);
-            assertEquals("name.?", arg.getName());
-            assertEquals("value_ here", arg.getValue());
-            assertEquals("name.%3F", arg.getEncodedName());
-            assertEquals("value_+here", arg.getEncodedValue());
-            // Show that can bypass encoding:
-            arg.setAlwaysEncoded(false);
-            assertEquals("name.?", arg.getEncodedName());
-            assertEquals("value_ here", arg.getEncodedValue());
-            
-            arg = new HTTPArgument("", "\00\01\07", "", false);
-            arg.setAlwaysEncoded(false);
-            assertEquals("", arg.getEncodedName());
-            assertEquals("\00\01\07", arg.getEncodedValue());
-        }
+	@Rule
```

Comment:
Please use spaces not tabs for indentation.

Reply:
Incorporated the review comments for changing the tabs to spaces

Reply:
Hi,

We had raised PR#440 which was closed due to migration of SVN>GIT.
So now we have raised a new PR#467 for  Bug#63130. 
Also all the comments which were raised for PR#440 have been incorporated in this new PR#467.
Kindly confirm those comments in this new PR#467.

Also, in PR#440, there was 1 comment (Mentioned below) related to encoding requirement for parameter name. 
"In this code you use a different encoding than UTF-8. Are you sure it's allowed for parameter names ? If yes, then the PR should impact getEncodedName() which uses UTF-8 by default".

As per our understanding & discussion with one of the Japanese developer, name is having a static value in the page and it has been used by the developers in the English only.  

So just want to have some input from other community members also.

Thank you!